compiler/src/Language/Bond/Codegen/Java/Class_java.hs

```diff
@@ -373,6 +373,41 @@ public #{declName}(com.microsoft.bond.StructBondType<#{declName}<#{paramList}>>
             else mempty
 
 
+-- builds text for Object.equals(Object) override
+object_equals :: String -> [Field] -> Text
+object_equals declName fields = [lt|@Override
+    public boolean equals(Object o) {
+        if (o == null) return false;
+        if (!(o instanceof #{declName})) return false;
+        final #{declName} other = (#{declName}) o;
+        #{newlineSep 2 compareField fields}
+        return true;
+    }
+|]
+    where
+        compareField Field {..} = [lt|if (!(#{equals_member fieldType fieldName})) return false;|]
+        equals_member t f
+            | isJavaPrimitive t = [lt|this.#{f} == other.#{f}|]
```

Comment:
I don't have strong opinion either way, as I am not convinced that we even need deep comparison and hash code on generated types. I wasn't there where this was decided; let's reiterate on this today.

If C++/C# don't have this feature, let's not add this to Java. It appears that C# doesn't have it.

If C++/C# have this feature, let's keep it consistent.

---
In reply to: [131471537](https://github.com/Microsoft/bond/pull/545#discussion_r131471537) [](ancestors = 131471537)